.github/workflows/typescript_test.yml

```diff
@@ -33,6 +33,61 @@ env:
   PLAYWRIGHT_BROWSERS_PATH: "ms-playwright"
 
 jobs:
+  determine-test-suite:
+    runs-on: ubuntu-latest
+    outputs:
+      suites: ${{ steps.set-matrix.outputs.matrix }}
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ inputs.ref || github.ref }}
+          fetch-depth: 0
+
+      - name: Paths Filter
+        id: filter
+        uses: dorny/paths-filter@v2
+        with:
+          filters: .github/filters.yml
+          base: ${{ github.event.before }}
+          head: ${{ github.sha }}
+
+      - name: Set Matrix
+        id: set-matrix
+      run: |
+        SUITES="[]"
+
+        # Check components changes (excluding starter projects)
+        if [[ "${{ steps.filter.outputs.components }}" == "true" ]]; then
+          SUITES=$(echo $SUITES | jq '. += ["components"]')
+        fi
+
+        # Check starter-projects changes (including components in starter projects)
+        if [[ "${{ steps.filter.outputs.starter-projects }}" == "true" ]]; then
+          SUITES=$(echo $SUITES | jq '. += ["starter-projects"]')
+        fi
+
+        # Check workspace changes
+        if [[ "${{ steps.filter.outputs.workspace }}" == "true" ]]; then
+          SUITES=$(echo $SUITES | jq '. += ["workspace"]')
+        fi
+
+        # Check API changes
+        if [[ "${{ steps.filter.outputs.api }}" == "true" ]]; then
+          SUITES=$(echo $SUITES | jq '. += ["api"]')
+        fi
+
+        # Check database changes
+        if [[ "${{ steps.filter.outputs.database }}" == "true" ]]; then
+          SUITES=$(echo $SUITES | jq '. += ["database"]')
+        fi
+
+        # If release changes or no specific changes detected, run all suites
+        if [[ "${{ steps.filter.outputs.release }}" == "true" || "$SUITES" == "[]" ]]; then
+          SUITES='["starter-projects", "components", "workspace", "api", "database"]'
+        fi
```

Comment:
This is better if we put it as an input from the workflow_call and workflow_dispatch